compiler/next/include/chpl/types/QualifiedType.h

```diff
@@ -41,18 +42,26 @@ class Type;
  */
 class QualifiedType final {
  public:
-  enum Kind {
-    UNKNOWN,
-    CONST,
-    REF,
-    CONST_REF,
-    VALUE,
-    CONST_VALUE,
-    TYPE,
-    PARAM,
-    FUNCTION,
-    MODULE,
-  };
+  using Kind = uast::IntentList;
+
+  // instead of the below,
+  // could use 'using enum uast::IntentList' in C++20
+  static const Kind UNKNOWN = uast::IntentList::UNKNOWN;
```

Comment:
What's the reason to bring all these out? It looks like the other enum Kind's are still an enum?

Reply:
Yes. I wanted to have code working with QualifiedType able to use e.g. `QualifiedType::TYPE`. At the same time, I wanted to be able to easily go between an `IntentList` and a `QualifiedType::Kind` (in both directions). This does not make sense in most of the other use cases where the local enum is a subset of IntentList.